python/packages/ai/teams/ai/promptsv2/user_input_message.py

```diff
@@ -0,0 +1,115 @@
+              Defaults to `input`.
+            files_variable (str, optional): Name of the variable containing the user input files.
+              Defaults to `inputFiles`.
+        """
+        super().__init__(tokens, True, "\n", "user: ")
+        self._input_variable = input_variable
+        self._files_variable = files_variable
+
+    async def render_as_messages(
+        self,
+        context: TurnContext,
+        memory: Memory,
+        functions: PromptFunctions,
+        tokenizer: Tokenizer,
+        max_tokens: int,
+    ) -> RenderedPromptSection[List[Message[Any]]]:
+        """
+        Renders the section as a list of messages.
+
+        Args:
+            context (TurnContext): Context for the current turn of conversation with the user.
+            memory (Memory): An interface for accessing state values.
+            functions (PromptFunctions): Registry of functions that can be used by the section.
+            tokenizer (Tokenizer): Tokenizer to use when rendering the section.
+            max_tokens (int): Maximum number of tokens allowed to be rendered.
+
+        Returns:
+            RenderedPromptSection[List[Message]]: The rendered prompt section as a list of messages.
+        """
+        # Get input text & images
+        input_text: str = memory.get_value(self._input_variable) or ""
+        input_files: List[InputFile] = memory.get_value(self._files_variable) or []
+
+        # Create message
+        message: Message[List[MessageContentParts]] = Message("user", [])
+
+        # Append text content part
+        length = 0
+        budget = self._get_token_budget(max_tokens)
+        if len(input_text) > 0:
+            encoded = tokenizer.encode(input_text)
+            if len(encoded) <= budget:
+                if message.content is not None:
+                    message.content.append(TextContentPart(type="text", text=input_text))
+                    length += len(encoded)
+                    budget -= len(encoded)
+            else:
+                if message.content is not None:
+                    message.content.append(
+                        TextContentPart(type="text", text=tokenizer.decode(encoded[:budget]))
+                    )
+                    length += budget
+                    budget = 0
+
+        # Append image content parts
+        images = [f for f in input_files if f.content_type.startswith("image/")]
+        for image in images:
+            # Check for budget to add image
+            if budget < 85:
+                break
+
+            # Add image
+            url = (
+                f"data:{image.content_type};base64,"
+                f"{base64.b64encode(image.content).decode('utf-8')}"
+            )
+            if message.content is not None:
+                message.content.append(ImageContentPart(type="image_url", image_url=ImageUrl(url)))
+                length += 85
+                budget -= 85
+
+        # Return output
```

Comment:
this comment seems obvious by the code so I don't think its necessary

Reply:
Followed the JS SDK to add the comprehensive comments. Will remove them.